components/sidebar/sidebar_list/sidebar_list.tsx

```diff
@@ -300,42 +309,39 @@ export default class SidebarChannelList extends React.PureComponent<Props, State
         return this.getFirstUnreadChannelFromChannelIdArray(this.getDisplayedChannelIds().reverse());
     }
 
-    navigateByChannelId = (id: string) => {
-        if (this.props.collapsedThreads && id === '') {
-            this.props.actions.switchToGlobalThreads();
-        } else {
+    navigateById = (id: string) => {
+        if (this.props.staticItems.findIndex((i) => i.id === id) === -1) {
             this.props.actions.switchToChannelById(id);
+        } else {
+            this.props.actions.switchToStaticItem(id);
         }
     }
 
     navigateChannelShortcut = (e: KeyboardEvent) => {
         if (e.altKey && !e.shiftKey && !e.ctrlKey && !e.metaKey && (Utils.isKeyPressed(e, Constants.KeyCodes.UP) || Utils.isKeyPressed(e, Constants.KeyCodes.DOWN))) {
             e.preventDefault();
 
-            const allChannelIds = this.getDisplayedChannelIds();
-            const curChannelId = this.props.currentChannelId;
+            const staticItemIds = this.getDisplayedLhsStaticItemIds();
+            const allIds = [...staticItemIds, ...this.getDisplayedChannelIds()];
 
-            if (this.props.collapsedThreads) {
-                // threads set channel id to ''
-                // add it to allChannelIds
-                allChannelIds.unshift('');
+            let curSelectedId = this.props.currentChannelId;
```

Comment:
This could be shortened to `const curSelectedId = this.props.currentChannelId || this.props.currentStaticItemId`.

Also, nothing to change here, but this along with renaming the file is making me think a bit about how we might make this more generic in the future since we have talked about making it a common component. Perhaps this part, for example, could be done more like our accessibility code where it uses the DOM more. For example, it could theoretically find which `.SidebarItem` is currently `.selected` and then trigger a click on the next/previous `.SidebarItem`. Just some thoughts though for the future.

Reply:
Also, I can't suggest this on the right line, but would you mind renaming the component to `SidebarList` to match the new file name?